src/lambda/trans.lisp

```diff
@@ -221,6 +233,99 @@ This follows from the fact that bool arapped in maybe is 1 + (bool + bool)"
+                                                         (rec context rtm))))))
+               ((lamb-eq ltm rtm) (let ((tyltm (ttype ltm)))
+                                    (comp (mcase (terminal (prod (maybe tyltm)
+                                                                 so1))
+                                                 (commutes-left (comp (mcase (terminal (prod tyltm
+                                                                                             so1))
+                                                                             (geb.extension.spec:nat-eq tyltm))
+                                                                      (distribute (ttype ltm)
+                                                                                  so1
+                                                                                  (ttype ltm)))))
+                                          (comp (distribute (maybe tyltm)
+                                                            so1
+                                                            (ttype ltm))
+                                                (geb:pair (rec context ltm)
+                                                          (rec context rtm))))))
+               ((lamb-lt ltm rtm) (let ((tyltm (ttype ltm)))
+                                    (comp (mcase (terminal (prod (maybe tyltm)
+                                                                 so1))
+                                                 (commutes-left (comp (mcase (terminal (prod tyltm
+                                                                                             so1))
+                                                                             (geb.extension.spec:nat-lt tyltm))
+                                                                      (distribute (ttype ltm)
+                                                                                  so1
+                                                                                  (ttype ltm)))))
+                                          (comp (distribute (maybe tyltm)
+                                                            so1
+                                                            (ttype ltm))
+                                                (geb:pair (rec context ltm)
+                                                          (rec context rtm))))))
+               ((bit-choice bitv)  (comp (comp (->right so1 (ttype tterm))
+                                              (nat-const (ttype tterm) (reduce
+                                                                        #'(lambda (a b)
+                                                                            (+ (ash a 1) b))
+                                                                        bitv)))
+                                        (terminal (stlc-ctx-maybe context))))
```

Comment:
I see lambda is still horrible as ever.

Let us attept to clean it up.

Please fix the formatting by newlining after `(bit-choice bitv)`, so the code it maps to is on the next line, given just how big they are.